arbiter/drivers/onedrive.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#ifndef ARBITER_IS_AMALGAMATION
+#include <arbiter/drivers/http.hpp>
+#endif
+
+#ifdef ARBITER_CUSTOM_NAMESPACE
+namespace ARBITER_CUSTOM_NAMESPACE
+{
+#endif
+
+namespace arbiter
+{
+
+namespace drivers
+{
+
+class OneDrive : public Https
+{
+    class Auth;
+    class Resource;
+
+public:
+    OneDrive(http::Pool& pool, std::unique_ptr<Auth> auth);
+
+    static std::unique_ptr<OneDrive> create(http::Pool& pool, std::string j);
+
+    virtual std::string type() const override { return "od"; };
+    virtual std::unique_ptr<std::size_t> tryGetSize(
+            std::string path) const override;
+private:
+    virtual bool get(
+        std::string path,
+        std::vector<char>& data,
+        http::Headers headers,
+        http::Query query) const override;
+
+    virtual std::vector<std::string> glob(
+            std::string path,
+            bool verbose) const override;
+
+    std::vector<std::string> processList(std::string path, bool recursive) const;
+
+    std::unique_ptr<Auth> m_auth;
+
+};
+
+class OneDrive::Auth
+{
+public:
+    Auth(std::string s);
+    static std::unique_ptr<Auth> create(std::string s);
+    std::string getToken() { return m_token; };
+    // static std::string getToken(std::string s);
+    std::string getRefreshUrl() { return m_authUrl + "/common/oauth2/v2.0/token"; }
```

Comment:
This function (and the member `m_authUrl`) seem to be constant - better off putting them in the unnamed namespace instead of as members.